Client/Frontend/Settings/Brave Today/BraveTodayAddSourceViewController.swift

```diff
@@ -0,0 +1,371 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Shared
+import BraveShared
+import BraveUI
+import Fuzi
+import FeedKit
+
+class BraveTodayAddSourceViewController: UITableViewController {
+    
+    let feedDataSource: FeedDataSource
+    var sourcesAdded: ((Set<RSSFeedLocation>) -> Void)?
+    
+    private var isLoading: Bool = false
+    
+    init(dataSource: FeedDataSource) {
+        self.feedDataSource = dataSource
+        if #available(iOS 13.0, *) {
+            super.init(style: .insetGrouped)
+        } else {
+            super.init(style: .grouped)
+        }
+    }
+    
+    @available(*, unavailable)
+    required init(coder: NSCoder) {
+        fatalError()
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        title = Strings.BraveToday.addSource
+        
+        navigationController?.navigationBar.prefersLargeTitles = true
+        navigationItem.largeTitleDisplayMode = .always
+        navigationItem.backButtonTitle = ""
+        navigationItem.leftBarButtonItem = .init(barButtonSystemItem: .cancel, target: self, action: #selector(tappedCancel))
+        
+        textField.addTarget(self, action: #selector(textFieldTextChanged), for: .editingChanged)
+        textField.delegate = self
+        
+        tableView.register(FeedSearchCellClass.self)
+        tableView.register(CenteredButtonCell.self)
+        tableView.tableHeaderView = UIView(frame: .init(x: 0, y: 0, width: 0, height: 10))
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        textField.becomeFirstResponder()
+    }
+    
+    @objc private func tappedCancel() {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @objc private func textFieldTextChanged() {
+        if let cell = tableView.cellForRow(at: IndexPath(row: 1, section: 0)) as? CenteredButtonCell {
+            // Update the color of the search row when text field is non empty
+            cell.tintColor = isSearchEnabled && !isLoading ? BraveUX.braveOrange : Colors.grey500
+        }
+    }
+    
+    private func tappedImportOPML() {
+        let picker = UIDocumentPickerViewController(documentTypes: ["public.opml"], in: .import)
+        picker.delegate = self
+        picker.allowsMultipleSelection = false
+        if #available(iOS 13.0, *) {
+            picker.shouldShowFileExtensions = true
+        }
+        present(picker, animated: true)
+    }
+    
+    private func rssLocationFromOPMLOutline(_ outline: OPML.Outline) -> RSSFeedLocation? {
+        guard let url = outline.xmlUrl?.asURL else { return nil }
+        return .init(title: outline.text, url: url)
+    }
+    
+    private let session: URLSession = {
+        let configuration = URLSessionConfiguration.default
```

Comment:
maybe `ephemeral` session would be better for this task? I think Brandon mentioned using it when fetching stuff for playlist, this is kind of similar